src/Build/Instance/ImmutableProjectCollections/ImmutableProjectMetadataCollectionConverter.cs

```diff
@@ -2,84 +2,114 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System;
+using System.Collections;
 using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
 using Microsoft.Build.Collections;
 using Microsoft.Build.Evaluation;
-using Microsoft.Build.Execution;
 using Microsoft.Build.Shared;
 
 namespace Microsoft.Build.Instance
 {
-    internal class ImmutableProjectMetadataCollectionConverter :
-        ImmutableElementCollectionConverter<ProjectMetadata, ProjectMetadataInstance>,
-        ICopyOnWritePropertyDictionary<ProjectMetadataInstance>
+    internal class ImmutableProjectMetadataCollectionConverter : IReadOnlyDictionary<string, string>
     {
         /// <summary>
         /// Immutable project item.
         /// </summary>
         private readonly ProjectItem _linkedProjectItem;
 
+        /// <summary>
+        /// Properties in the underlying dictionary.
+        /// This dictionary contains all properties that are directly defined in the project item,
```

Comment:
Nit: This sentence ends with a comma instead of a period.

Reply:
thanks, fixed now.